Classes/Infrastructure/ContentRepository/CreationDialog/PromotedElementsCreationHandlerFactory.php

```diff
@@ -56,24 +55,19 @@ public function handle(NodeCreationCommands $commands, NodeCreationElements $ele
                     if ($nodeType->hasReference($elementName)) {
                         assert($elementValue instanceof NodeAggregateIds);
                         $referenceConfiguration = $nodeType->getReferences()[$elementName];
-                        if (
-                            ($referenceConfiguration['ui']['showInCreationDialog'] ?? false) === true
-                        ) {
-                            // a promoted element
-                            $setReferencesCommands[] = SetNodeReferences::create(
-                                $commands->first->workspaceName,
-                                $commands->first->nodeAggregateId,
-                                $commands->first->originDimensionSpacePoint,
-                                ReferenceName::fromString($elementName),
-                                NodeReferencesToWrite::fromNodeAggregateIds($elementValue)
-                            );
+                        if (($referenceConfiguration['ui']['showInCreationDialog'] ?? false) === true) {
+                            $referencesToWriteForCurrentElement = NodeReferencesToWrite::fromNameAndTargets(ReferenceName::fromString($elementName), $elementValue);
+                            $initialReferences = $initialReferences === null ? $referencesToWriteForCurrentElement : $initialReferences->merge($referencesToWriteForCurrentElement);
```

Comment:
i would just always merge here and initialise initialReferences with empty `NodeReferencesToWrite` and then let the core optimise this away :D